Add Release Timeline to community-plugins.json

cakechaser opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/cakechaser/obsidian-release-timeline

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

liamcain commented:
Really cool plugin!

---

Since this plugin requires dataview to be installed, I recommend also adding a check and warning the user that it's a requirement. Otherwise users might run into unexpected errors. Maybe something like:

```
const isDataviewInstalled = !!getAPI();
if (!isDataviewInstalled) {
  new Notice("The . The Release-Timeline plugin requires Dataview to properly function.");
}
```

Just make sure not to put it directly in the `onLoad` is it's not guaranteed this plugin will be loaded after dataview. Inside `onLayoutReady` should work though.

https://github.com/cakechaser/obsidian-release-timeline/blob/10251e87b0303e9e405d08c9d9ed1da337a0a96d/main.ts#L28 You don't want to be querying the whole document here. I would suggest changing this to:

```ts
let matches = el.querySelectorAll(".td-first, .td-next");
```

https://github.com/cakechaser/obsidian-release-timeline/blob/10251e87b0303e9e405d08c9d9ed1da337a0a96d/functions.ts#L6 This is really just a matter of code style, but I would recommend not wrapping your helper functions inside a class. You can just keep them as top-level functions and pass in the plugin instance to the functions that need it. Reason being:
- You might want to split this file into different files based on purpose at some point (`parse.ts`, `timelineRender.ts` for example)
- These functions are less portable if the plugin gets bigger and you need to use the function in different parts of the code.
- They are slightly harder to test, if you wanted to add unit tests down the road

Anyway, not suggesting you need to refactor it, but wanted to offer some unsolicited advice!

cakechaser commented:
Hi @liamcain, thank you very much for the feedback!

I've added the Dataview notice and fixed the html query from your 2nd suggestion.

Tried to refactor the helper functions from your 3rd suggestion, but ran into some issues, and it was taking a long time for me to figure this out, since it's my first JS project.
So I left them as is for now, hopefully it's not an issue. Nevertheless, thank you very much for the suggestion, I'll try to refactor them for one of the next releases.

liamcain reviewed (approved):
Looks great! Nice job. This will be merged in shortly.
